pkg/containerutil/cp_linux.go

```diff
@@ -247,3 +275,46 @@ func mountSnapshotForContainer(ctx context.Context, client *containerd.Client, c
 	}
 	return tempDir, cleanup, nil
 }
+
+func getContainerMountInfo(ctx context.Context, con containerd.Container, containerPath string) (string, string, error) {
+	filePath := filepath.Clean(containerPath)
+	spec, err := con.Spec(ctx)
+	if err != nil {
+		return "", "", err
+	}
+	if spec.Root.Readonly {
+		return "", "", fmt.Errorf("container rootfs: %s is marked read-only", spec.Root.Path)
+	}
+
+	for _, mount := range spec.Mounts {
+		if isParentDir(filePath, mount.Destination) {
+			for _, option := range mount.Options {
+				if option == "ro" {
+					return "", "", fmt.Errorf("mount point %s is marked read-only", filePath)
+				}
+			}
+			return mount.Source, mount.Destination, nil
+		}
+	}
+	return "", "", nil
+}
+
+func isParentDir(filePath, potentialParent string) bool {
+	filePath = filepath.Clean(filePath)
+	potentialParent = filepath.Clean(potentialParent)
+	if filePath == potentialParent {
```

Comment:
`/etc/var/test` is not a parent of `/etc/var/test`
```suggestion
	if path.Dir(filePath) == potentialParent
```

Reply:
We want isParentDir("/etc/var/test", "/etc/var/test") to return `true` since `/etc/var/test` is in the volume so is `/etc/var/test/test1`. I will rename the function to reflect that.